tests/Doctrine/Tests/DBAL/Functional/BlobTest.php

```diff
@@ -55,6 +57,43 @@ public function testInsert()
         self::assertEquals(1, $ret);
     }
 
+    public function testInsertProcessesStream()
+    {
+        $this->_conn->insert('blob_table', [
+            'id'          => 1,
+            'clobfield'   => 'ignored',
+            'blobfield'   => fopen('data://text/plain,test', 'r'),
+            'binaryfield' => fopen('data://text/plain,test', 'r'),
```

Comment:
> The `binary` and `blob` types map `BLOB`/`BINARY`/`VARBINARY` columns to PHP streams.

It's not exactly true (see https://github.com/doctrine/dbal/commit/5465c6c90fb41bd9f040716f902f015abb710453). Only BLOB and CLOB (TEXT) fields should be populated from streams.

Reply:
I think I need some more explications here 🤔

Please have a look at the [current test on 2.7](https://github.com/doctrine/dbal/blob/9efbf7f3e260442ceefe4aec163c18b193c8e0c2/tests/Doctrine/Tests/DBAL/Functional/BlobTest.php):

* `clobfield` is always used with `ParameterType::STRING`. It does not return a stream on `SELECT`, so I also did not expect streams to be bound here (could change that).
* `blobfield` is a `ParameterType::LARGE_OBJECT`, should process streams when binding data and return streams from `SELECT` (as-is right now).
* `binaryfield` is always treated like `blobfield`, including the [assertions that streams are returned](https://github.com/doctrine/dbal/blob/9efbf7f3e260442ceefe4aec163c18b193c8e0c2/tests/Doctrine/Tests/DBAL/Functional/BlobTest.php#L102). Only difference is that [its database type is `binary`](https://github.com/doctrine/dbal/blob/9efbf7f3e260442ceefe4aec163c18b193c8e0c2/tests/Doctrine/Tests/DBAL/Functional/BlobTest.php#L30).

Your remarks would make sense to me if `clobfield` were treated as `ParameterType::LARGE_OBJECT` and `binaryfield` were `ParameterType::STRING`, because then stream processing (on read as well as write) would only happen for the `LARGE_OBJECT` type.

Reply:
Your remark regarding changed behaviour is true for 3.x, but this is against 2.7.